src/modules/fancyzones/lib/FancyZonesData.cpp

```diff
@@ -70,7 +74,21 @@ FancyZonesData& FancyZonesDataInstance()
 
 FancyZonesData::FancyZonesData()
 {
-    std::wstring saveFolderPath = PTSettingsHelper::get_module_save_folder_location(NonLocalizable::FancyZonesStr);
+    std::wstring saveFolderPath{};
+    try
+    {
+        saveFolderPath = PTSettingsHelper::get_module_save_folder_location(NonLocalizable::FancyZonesStr);
+    }
+    catch (...)
+    {
+        // Not being able to find path to persisted data completely breaks down FancyZones functionality.
+        std::wstring errorMessage = GET_RESOURCE_STRING(IDS_FANCYZONES_DATA_ERROR) + L" " + NonLocalizable::PowerToysIssuesURL;
+        MessageBox(NULL,
+                   errorMessage.c_str(),
+                   GET_RESOURCE_STRING(IDS_FANCYZONES).c_str(),
+                   MB_OK | MB_ICONERROR);
+        exit(1);
```

Comment:
It would be better if we can just disable FZ and not exit the runner otherwise it would be impossible to turn FZ off for the user and still use the other modules.

Reply:
That is correct. However I'm not sure about mechanism to signalize from library back to dll to trigger disable.

Reply:
FancyZones disabled in case of unrecoverable error.